changes.go

```diff
@@ -780,8 +780,15 @@ func (s *ChangesService) change(tail string, changeID string, input interface{})
 
 	v := new(ChangeInfo)
 	resp, err := s.client.Do(req, v)
+	if err != nil {
+		return v, resp, err
+	}
 	if resp.StatusCode == http.StatusConflict {
-		body, _ := ioutil.ReadAll(resp.Body)
+		var body []byte
+		body, err = ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return v, resp, err
+		}
 		err = errors.New(string(body[:]))
 	}
 	return v, resp, err
```

Comment:
I would suggest rewriting as follows to make the code more readable:

```Go
...
if resp.StatusCode == http.StatusConflict {
	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		return v, resp, err
	}
	return v, resp, errors.New(string(body[:]))
}
return v, resp, nil
```

Also, I'm suspect about this whole thing working. Doesn't `s.client.Do(req, v)` already consume the `resp.Body` before returning? There are no tests for it, and no rationale for why it was added in #15. I'd remove it and simplify this endpoint to be like all others. But it's out of scope of this PR.

Reply:
I adjusted the code according to your suggestion.

Regarding your comment: Indeed. That could be a possible mistake. And I am with you, that this has to be investigated in a separate ticket/pr. Let me check this, and I will ping you for a review again.

Reply:
Hey @dmitshur, i am a little bit packed with private things the next days/weeks. I will come back to this later. If you have some spare time around, feel free to make the change. Otherwise i can do it later. Sorry, but thanks for understanding.

Reply:
Not at a problem at all, this isn't blocking me in any way.

If you don't want this opportunity to improve code to be forgotten, I'd suggest opening an issue to track it.

Reply:
Just to not forget it: https://github.com/andygrunwald/go-gerrit/issues/64 (was short on time, but i think it is enough for us now).